src/CP/valueselection/learning/rewards/generalreward.jl

```diff
@@ -0,0 +1,122 @@
+"""
+    struct GeneralReward <: AbstractReward end
+
+This is a general reward encouraging a smart exploration of the tree in the case DFSearch is used.
+"""
+mutable struct GeneralReward <: AbstractReward 
+    value::Float32
+    initMin::Union{Nothing,Int}
+    initMax::Union{Nothing,Int}
+    initialNumberOfVariableValueLinks::Int
+    gamma::Float32
+    beta::Float32
+end
+
+function GeneralReward(model::CPModel)
+    if !isnothing(model.objective)
+        return GeneralReward(0, model.objective.domain.min.value, model.objective.domain.max.value, global_domain_cardinality(model), 2.0, 2.0)
+    else
+        return GeneralReward(0, nothing, nothing, global_domain_cardinality(model), 2.0, 2.0)
+    end
+end
+
+"""
+    set_reward!(::StepPhase, lh::LearnedHeuristic{GeneralReward, A}, model::CPModel, symbol::Union{Nothing, Symbol})
+
+Change the "current_reward" attribute of the LearnedHeuristic at the StepPhase.
+"""
+function set_reward!(::Type{StepPhase}, lh::LearnedHeuristic{SR, GeneralReward, A}, model::CPModel, symbol::Union{Nothing, Symbol}) where {
+    SR <: AbstractStateRepresentation,
+    A <: ActionOutput
+}
+    nothing
+end
+
+"""
+    set_reward!(::DecisionPhase, lh::LearnedHeuristic{GeneralReward, O}, model::CPModel)
+
+Change the current reward at the DecisionPhase. This is called right before making the next decision, so you know you have the very last state before the new decision
+and every computation like fixPoints and backtracking has been done.
+"""
+function set_reward!(::Type{DecisionPhase}, lh::LearnedHeuristic{SR, GeneralReward, A}, model::CPModel) where {
+    SR <: AbstractStateRepresentation,
+    A <: ActionOutput
+}
+    if !lh.firstActionTaken
+        #println("First action being taken")
+        lh.reward.value = 0
+    else
+        lh.reward.value = lh.reward.gamma*(model.statistics.lastPruning/(lh.reward.initialNumberOfVariableValueLinks - length(branchable_variables(model))))^lh.reward.beta
+        #println(model.statistics.lastPruning)
+        #println(lh.reward.initialNumberOfVariableValueLinks - length(branchable_variables(model)))
+        #println(string(lh.reward.value))
+        if lh.reward.value>0.001
```

Comment:
This `if` is now unnecessary.

Reply:
Was used for debugging only. All the instructions and comments used for debugging are now removed.